index.js

```diff
@@ -0,0 +1,248 @@
+let size = 4;
+let min = 0;
+let max = size - 1;
+let isMoved = false;
+let score = 0;
+let excludeIds = [];
+const loadTable = () => {
+  document.getElementById("canvas_table").innerHTML = "";
+  const table = document.createElement("table");
+  const tbody = document.createElement("tbody");
+  for (let r = 0; r < size; r++) {
+    const row = document.createElement("tr");
+    for (let c = 0; c < size; c++) {
+      let id = `${r},${c}`;
+      const column = document.createElement("td");
+      column.setAttribute("id", id);
+      row.appendChild(column);
+    }
+    tbody.appendChild(row);
+  }
+  table.appendChild(tbody);
+  document.getElementById("canvas_table").appendChild(table);
+
+  let id1 = getId();
+  let id2 = "";
+  while (true) {
+    id2 = getId();
+    if (id1 != id2) break;
+  }
+
+  const box1 = document.getElementById(id1);
+  const box2 = document.getElementById(id2);
+
+  box1.innerHTML = 2;
+  box1.style.backgroundColor = getColor(2);
+
+  box2.innerHTML = 2;
+  box2.style.backgroundColor = getColor(2);
+
+  score = 0;
+  document.getElementById("score").innerHTML = score;
+
+  return false;
+};
+
+const getRandomId = () => {
+  return Math.floor(Math.random() * (max - min + 1) + min);
+};
+
+const getId = () => {
+  let r = getRandomId();
+  let c = getRandomId();
+  return `${r},${c}`;
+};
+
+const getColor = (val = 0) => {
+  const colors = {
+    0: "#ffffff",
+    2: "#fd7f6f",
+    4: "#7eb0d5",
+    8: "#b2e061",
+    16: "#bd7ebe",
+    32: "#ffb55a",
+    64: "#ffee65",
+    128: "#beb9db",
+    256: "#fdcce5",
+    512: "#8bd3c7",
+    1024: "#8000ff",
+    2048: "#ff00ff",
+  };
+  return colors[val];
+};
+
+document.onkeydown = (e) => {
+  e.preventDefault();
+  switch (e.keyCode) {
+    case 37:
+      getKeyDown("left");
+      break;
+    case 38:
+      getKeyDown("up");
+      break;
+    case 39:
+      getKeyDown("right");
+      break;
+    case 40:
+      getKeyDown("down");
+      break;
+  }
+};
+
+const checkNotNull = (keyValue) => {
+  isMoved = false;
+  excludeIds = [];
+  for (let r = min; r <= max; r++) {
+    for (let c = min; c <= max; c++) {
+      let id = `${r},${c}`;
+      if (document.getElementById(id).innerHTML !== "") {
+        checkMove(id, keyValue);
+      }
+    }
+  }
```

Comment:
this loop not support function keydown right and down 

- right keydown must check from right column to left column
- down keydown must check bottom row to top row